File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
import heapq
from collections import Counter

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = Counter(nums)
        heap = []
        for num, count in counter.items():
            heapq.heappush(heap, (-count, num))
        res = []
        while k > 0:
            res.append(heapq.heappop(heap)[1])
            k -= 1
        return res
        

File: Data Structures & Algorithms/majority-element/submission-3.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 1;
        int num = nums[0];

        for (int i=1;i<nums.size();i++) {
            if (num == nums[i]) count += 1;
            else {
                count -= 1;
                if (count == 0) {
                    count = 1;
                    num = nums[i];
                }
            }
        }
        return num;
    }
};